src/kafka-file/languageservice/services/diagnostics.ts

```diff
@@ -253,4 +262,31 @@ export class KafkaFileDiagnostics {
             }
         }
     }
+
+    async validateTopic(topicProperty: Property | undefined, blockType: BlockType, diagnostics: Diagnostic[]) {
+        if (!topicProperty) {
+            return;
+        }
+        const topicId = topicProperty.value?.content.trim();
+        if (!topicId || topicId.length < 1) {
+            return;
+        }
+        const { clusterId } = this.selectedClusterProvider.getSelectedCluster();
+        if (clusterId) {
+            if (!await this.topicProvider.getTopic(clusterId, topicId)) {
+                const range = topicProperty.propertyTrimmedValueRange || topicProperty.propertyRange;
+                const autoCreate = await this.topicProvider.isAutoCreateTopicEnabled(clusterId);
+                const errorMessage = getTopicErrorMessage(topicId, autoCreate, blockType);
+                const severity = autoCreate ? DiagnosticSeverity.Warning : DiagnosticSeverity.Error;
+                diagnostics.push(new Diagnostic(range, errorMessage, severity));
+            }
+        }
+    }
 }
+
+function getTopicErrorMessage(topicId: string, autoCreate: boolean, blockType: BlockType): string {

```

Comment:
autocreate:enabled: [info] 'Topic will be created automatically' 
autocreate:disabled: [error] 'Cluster does not support automatic topic creation' 
autocreate:unknown: =>[warning] 'Cluster might not support automatic topic creation'

Reply:
fixed